relay-cardinality/src/limiter.rs

```diff
@@ -141,10 +139,15 @@ impl<T: Limiter> CardinalityLimiter<T> {
 #[derive(Debug)]
 pub struct CardinalityLimits<T> {
     source: Vec<T>,
-    rejections: HashSet<usize>,
+    rejections: BTreeSet<usize>,
 }
 
 impl<T> CardinalityLimits<T> {
+    /// Returns an iterator yielding only rejected items.
+    pub fn rejected(&self) -> impl Iterator<Item = &T> {
```

Comment:
Is this only for tests?
```suggestion
    /// Returns an iterator yielding only rejected items.
    #[cfg(test)]
    pub fn rejected(&self) -> impl Iterator<Item = &T> {
```

Reply:
No, only rejections are turned into outcomes. Only a subset of buckets may be cardinality limited while the rest is passed on.